Reqnroll/Infrastructure/MatchArgumentCalculator.cs

```diff
@@ -4,6 +4,12 @@
 
 namespace Reqnroll.Infrastructure
 {
+    // The ArgumentInfo structure holds information about arguments extracted from a Step match. The StartOffset indicates where in the step text the value of the argument begins in that string.
+    public class ArgumentInfo
+    {
+        public object Value;
+        public int StartOffset;
+    }
     public interface IMatchArgumentCalculator
     {
         object[] CalculateArguments(Match match, StepInstance stepInstance, IStepDefinitionBinding stepDefinitionBinding);
```

Comment:
Change the return type from `object[]` to `MatchArgument[]`, so that it becomes clear from the implementors what we expect.